File: Ex-25/ex_25_python/CalcDados.py
class CalcDados:
    def __init__(self):
        self.n1 = 0
        self.n2 = 0
        self.operacao = '\0'

    def setOperando(self, i, valor):
        if i == 1:
            self.n1 = valor
        elif i == 2:
            self.n2 = valor
        else:
            pass

    def getOperando(self, i):
        if i == 1:
            return self.n1
        elif i == 2:
            return self.n2
        else:
            print("Operando " + str(i) + " inválido")
            exit(1)

    def setOperacao(self, operacao):
        self.operacao = operacao

    def getOperacao(self):
        return self.operacao

    def reset(self):
        self.n1 = 0
        self.n2 = 0
        self.operacao = '\0'

File: Ex-25/ex_25_java/CalcDados.java
public class CalcDados {
    private double n1, n2;
    private char operacao;

    public CalcDados() {
        this.n1 = 0;
        this.n2 = 0;
        this.operacao = '\0';
    }

    public void setOperando(int i, double valor) {
        switch (i) {
        case 1:
            this.n1 = valor;
            break;
        case 2:
            this.n2 = valor;
            break;
        default:
            break;
        }
    }

    public double getOperando(int i) {
        switch (i) {
        case 1:
            return this.n1;
        case 2:
            return this.n2;
        default:
            System.out.println("Operando " + i + " inválido");
            System.exit(1);
            return 0;
        }
    }

    public void setOperacao(char operacao) {
        this.operacao = operacao;
    }

    public char getOperacao() {
        return this.operacao;
    }

    public void reset() {
        this.n1 = 0;
        this.n2 = 0;
        this.operacao = '\0';
    }
}
